File: pkg/codegen/testing/test/testdata/discriminated-unions-pp/dotnet/discriminated-unions.cs
using Pulumi;
using AzureNative = Pulumi.AzureNative;

class MyStack : Stack
{
    public MyStack()
    {
        var databaseAccount = new AzureNative.DocumentDB.DatabaseAccount("databaseAccount", new AzureNative.DocumentDB.DatabaseAccountArgs
        {
            AccountName = "ddb1",
            ApiProperties = new AzureNative.DocumentDB.Inputs.ApiPropertiesArgs
            {
                ServerVersion = "3.2",
            },
            BackupPolicy = 
            {
                { "periodicModeProperties", new AzureNative.DocumentDB.Inputs.PeriodicModePropertiesArgs
                {
                    BackupIntervalInMinutes = 240,
                    BackupRetentionIntervalInHours = 8,
                } },
                { "type", "Periodic" },
            },
            DatabaseAccountOfferType = AzureNative.DocumentDB.DatabaseAccountOfferType.Standard,
            Locations = 
            {
                new AzureNative.DocumentDB.Inputs.LocationArgs
                {
                    FailoverPriority = 0,
                    IsZoneRedundant = false,
                    LocationName = "sourthcentralus",
                },
            },
            ResourceGroupName = "rg1",
        });
    }

}

File: pkg/codegen/testing/test/testdata/discriminated-unions-pp/python/discriminated-unions.py
import pulumi
import pulumi_azure_native as azure_native

database_account = azure_native.documentdb.DatabaseAccount("databaseAccount",
    account_name="ddb1",
    api_properties=azure_native.documentdb.ApiPropertiesArgs(
        server_version="3.2",
    ),
    backup_policy=azure_native.documentdb.PeriodicModeBackupPolicyArgs(
        periodic_mode_properties=azure_native.documentdb.PeriodicModePropertiesArgs(
            backup_interval_in_minutes=240,
            backup_retention_interval_in_hours=8,
        ),
        type="Periodic",
    ),
    database_account_offer_type=azure_native.documentdb.DatabaseAccountOfferType.STANDARD,
    locations=[azure_native.documentdb.LocationArgs(
        failover_priority=0,
        is_zone_redundant=False,
        location_name="sourthcentralus",
    )],
    resource_group_name="rg1")
